react-app/src/utils/formOrigin.test.ts

```diff
@@ -0,0 +1,103 @@
+// formOrigin.test.ts
+import { describe, it, expect, beforeEach, afterEach, vi } from 'vitest';
+import { getFormOrigin, ORIGIN, DASHBOARD_ORIGIN, DETAILS_ORIGIN, SPA_SUBMISSION_ORIGIN, WAIVER_SUBMISSION_ORIGIN } from './formOrigin'; 
+import { Authority } from 'shared-types/authority';
+import { getDashboardTabForAuthority } from './crumbs'; 
+
+// Mock the getDashboardTabForAuthority function
+vi.mock('./crumbs', () => ({
+  getDashboardTabForAuthority: vi.fn(),
+}));
+
+describe('getFormOrigin', () => {
+  let originalLocation: Location;
+
+  beforeEach(() => {
+    // Store the original location object
+    originalLocation = window.location;
+
+    // Create a mock location object
+    const mockLocation = {
+      ...originalLocation,
+      search: '',
+      assign: vi.fn(),
+      reload: vi.fn(),
+      replace: vi.fn(),
+    };
+
+    // Use Object.defineProperty to replace window.location with the mock object
+    Object.defineProperty(window, 'location', {
+      value: mockLocation,
+      writable: true,
+    });
+  });
+
+  afterEach(() => {
+    // Restore the original location object after each test
+    Object.defineProperty(window, 'location', {
+      value: originalLocation,
+      writable: true,
+    });
+    vi.clearAllMocks(); // Clear mocks after each test
+  });
+
+  it('should return the correct pathname and search for DETAILS_ORIGIN', () => {
+    window.location.search = `?${ORIGIN}=${DETAILS_ORIGIN}`;
+
+    const authority: Authority = Authority.MED_SPA;
+    const id = '12345';
+
+    // Call the function
+    const result = getFormOrigin({ id, authority });
+
+    expect(result).toEqual({
+      pathname: `/${DETAILS_ORIGIN}/${encodeURIComponent(authority)}/${encodeURIComponent(id)}`,
```

Comment:
Small nit: you wouldn't need the `encodeURIComponent` for testing

Reply:
see comment below for this test

Reply:
In regards to your comment [here](https://github.com/Enterprise-CMCS/macpro-mako/pull/829/files/57e8d76c53d01783bd5fbcc21b54fb9b0023d02f#r1831356269):
> without mocking it the "should return the correct pathname and search for DASHBOARD_ORIGIN" returns false

I think the reason it's breaking is because `authority` is using the wrong `authority` the function expects. There's difference between `Authority.MED_SPA` and `Medicaid SPA`. So if you update `getFormOrigin` to use the correct expected `authority`, the test should pass for you, and you can also delete the mock at the top of the file

Reply:
I tried without mocking it and kept getting a failed dashboard and details origin test. Mocking getDashboardTabForAuthority allows it to control its output ('spas' or 'waivers') based on the test needs. This bypasses the internal check and allows it to use any value for authority in the test.

Reply:
What happens when you call the function with `getFormOrigin({ authority: 'Medicaid SPA' })